core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -54,23 +54,27 @@ object KafkaRequestHandler {
   }
 
   /**
-   * Wrap callback to schedule it on a request thread.
-   * NOTE: this function must be called on a request thread.
-   * @param fun Callback function to execute
-   * @return Wrapped callback that would execute `fun` on a request thread
+   * Creates a wrapped callback to be executed asynchronously on an arbitrary request thread.
+   * NOTE: this function must be originally called from a request thread.
+   * @param asyncCompletionCallback A callback method that we intend to call from the current thread or in another
```

Comment:
Should we also adjust "A callback method that we intend to call from the current thread" ?